From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Adam Lackorzynski <adam@os.inf.tu-dresden.de>
Date: Mon, 13 Aug 2018 14:39:48 +0200
Subject: [PATCH] loadconv: Add missing fclose()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Issue found and reported by David Binderman.

This is a backport of commit f66b5c78.

Signed-off-by: Ondřej Lysoněk <olysonek@redhat.com>
---
 src/config.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/config.c b/src/config.c
index 36b6e37..0da4989 100644
--- a/src/config.c
+++ b/src/config.c
@@ -1865,6 +1865,7 @@ void doconv(void)
 int loadconv(char *buf)
 {
   FILE *fp;
+  int err = 0;
 
   if ((fp = fopen(pfix_home(buf), "rb")) == (FILE *)NULL) {
     werror(_("Cannot open conversion table %s"), pfix_home(buf));
@@ -1874,10 +1875,11 @@ int loadconv(char *buf)
       || fread(vt_outmap, sizeof(vt_outmap), (size_t)1, fp) != 1)
     {
       werror(_("Cannot read conversion table %s"), pfix_home(buf));
-      return 1;
+      err = 1;
     }
+
   fclose(fp);
-  return 0;
+  return err;
 }
 
 int saveconv(char *buf)
